src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -414,6 +414,23 @@ def delete(self, request_name):
         # delete should also happen on WMStats
         cherrypy.log("INFO: Delete '%s' done." % request_name)
         
+    def _update_additional_request_args(self, workload, request_args):
+        """
+        add to request_args properties which is not initially set from user.
+        This data will put in to couchdb. 
+        Update request_args here if additional information need to be put in couchdb
+        """
+        request_args['RequestWorkflow'] = sanitizeURL("%s/%s/%s/spec" % (request_args["CouchURL"], 
+                                            request_args["CouchWorkloadDBName"], workload.name()))['url']
+            
+        request_args['OutputDatasets'] = []
+        # Add the output datasets if necessary
+        # for some bizarre reason OutpuDatasets is list of lists
+        for ds in workload.listOutputDatasets():
+            request_args['OutputDatasets'].append(ds)
```

Comment:
Thanks Alan, acutally can just do following (I have to check what exactly listOutputDatasets() returns but it seems it returns list of str.
```
request_args['OutputDatasets'] = workload.listOutputDatasets()
```